plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -52,102 +53,152 @@ def associate(source_repo, dest_repo, import_conduit, config, units=None, solver
     # get config items that we care about
     recursive = config.get(constants.CONFIG_RECURSIVE) or \
         config.get(constants.CONFIG_RECURSIVE_CONSERVATIVE)
+    additional_repos = config.get(constants.CONFIG_ADDITIONAL_REPOS, {})
+    if additional_repos and not recursive:
+        # TODO: is there a better error to raise?
+        raise ValueError("Cannot use additional_repos without one of the recursive flags set")
+
+    repo_map = additional_repos
+    repo_map[source_repo.repo_id] = dest_repo.repo_id
 
     if recursive and not solver:
         solver = pulp_solv.Solver(
             source_repo,
             target_repo=dest_repo,
             conservative=config.get(constants.CONFIG_RECURSIVE_CONSERVATIVE),
+            additional_repos=additional_repos,
             ignore_missing=False
             # the line above disables the code which injects "dummy solvables" to provide
             # missing packages. it is not known whether that code is 100% necessary, but it
             # is feeding invalid data to libsolv somehow resulting in a violated invariant
             # and failure on an assert statement here
             # https://github.com/openSUSE/libsolv/blob/master/src/solver.c#L1979-L1981
         )
+        # Only loads the original source and destination repositories
         solver.load()
 
-    # make a set from generator to be able to iterate through it several times
-    units = set(units)
-    failed_units = set()
-    # we need to filter out rpm units since in reality they were not associated
-    associated_units = [_associate_unit(dest_repo, unit, config) for unit in units
-                        if not isinstance(unit, models.RPM)]
-    if None in associated_units:
-        _LOGGER.warning(_('%s packages failed signature filter and were not imported.'
-                        % len(associated_units)))
-
-    associated_units = set(associated_units)
+    # make a list from generator to be able to iterate through it several times
+    units = list(units)
```

Comment:
any reason you changed this from set yo list so then you make is set again on line 116, am i missing something?

Reply:
It got cleaned up in a later commit